.github/workflows/ci.yml

```diff
@@ -96,43 +106,97 @@ jobs:
-        uses: actions/upload-artifact@v2
-        with:
-          name: artifact
-          path: dist/wgpu-*.zip
 
-  publish:
-    if: contains(github.ref, 'tags/v')
-    needs: [build]
-    name: Publish release artifacts
-    runs-on: ubuntu-18.04
+  # Builds for binary releases.
+  # Also build on non-tag commits; it also functions as a test.
+  # The Linux builds are performed on a "manylinux2010" container. This container
+  # is designed such that that the resulting binary has minimal dependencies on system
+  # libraries, and thus works on as many linuxes as possible. It's a thing from the
+  # Python world, but generally useful.
+  release-build:
+    needs: prepare
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        include:
+          - name: Linux-64
+            os: ubuntu-18.04
+            RUST_TOOLCHAIN: stable
+            ARCH: 64
+            OS_NAME: linux
+            IMAGE: manylinux2010_x86_64
+          - name: Linux-32
+            os: ubuntu-18.04
+            RUST_TOOLCHAIN: stable-i686-unknown-linux-gnu
+            ARCH: 32
+            OS_NAME: linux
+            IMAGE: manylinux2010_i686
+          - name: MacOS-64
+            os: macOS-10.14
+            RUST_TOOLCHAIN: stable
+            ARCH: 64
+            OS_NAME: macos
+            MACOSX_DEPLOYMENT_TARGET: '10.13'
+          - name: Windows-64
+            os: vs2017-win2016
+            RUST_TOOLCHAIN: stable-msvc
+            ARCH: 64
+            OS_NAME: windows
+          - name: Windows-32
+            os: vs2017-win2016
+            RUST_TOOLCHAIN: stable-i686-pc-windows-msvc
+            ARCH: 32
+            OS_NAME: windows
     steps:
-      - name: Download Artifacts
-        uses: actions/download-artifact@v2
-        with:
-          name: artifact
-          path: $GITHUB_WORKSPACE/wgpu-release/
-      - name: Create Release
-        id: create_release
-        uses: actions/create-release@v1
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        with:
-          tag_name: ${{ github.ref }}
-          release_name: Release ${{ github.ref }}
-          # TODO
-          body:
-          draft: false
-          prerelease: false
-      - name: Upload Release Asset
-        id: upload-release-asset
-        # Move back to official action after fix https://github.com/actions/upload-release-asset/issues/4
-        uses: csexton/release-asset-action@v2
-        with:
-          pattern: "$GITHUB_WORKSPACE/wgpu-release/*.zip"
-          release-url: ${{ steps.create_release.outputs.upload_url }}
-          github-token: ${{ secrets.GITHUB_TOKEN }}
+    - uses: actions/checkout@v2
+    - name: Docker build
+      if: success() && matrix.OS_NAME == 'Linux'
+      env:
+        IMAGE: ${{ matrix.IMAGE }}
+        RUST_TOOLCHAIN: ${{ matrix.RUST_TOOLCHAIN }}
+      run: |
+          CID=$(docker create -t -w /tmp/wgpu-native -v $PWD:/tmp/src:ro quay.io/pypa/$IMAGE bash -c "\
```

Comment:
Don't you need to preface these bash scripts with `set -ex`? Or is that by default on github actions?

Reply:
The default is `set -e` (-like behavior), which can be turned off with `fail-fast: false`

Reply:
Ok, that's fine then